src/js/gutenberg-plugins/content-resizing-plugin.js

```diff
@@ -0,0 +1,569 @@
+		if ( resizingType ) {
+			( async () => {
+				await resizeContent();
+			} )();
+		}
+	}, [ resizingType ] );
+
+	// Triggers AJAX request to resize the content.
+	useEffect( () => {
+		if ( isResizing ) {
+			( async () => {
+				const __textArray = await getResizedContent();
+				setTextArray( __textArray );
+				setIsModalOpen( true );
+			} )();
+		}
+	}, [ isResizing, getResizedContent ] );
+
+	// Resets to default states.
+	function resetStates() {
+		setSelectedBlock( null );
+		setTextArray( [] );
+		setIsModalOpen( false );
+		dispatch( resizeContentStore ).setResizingType( null );
+		isProcessAnimating = false;
+	}
+
+	/**
+	 * Triggered when either `Grow content` or `Shrink content` is clicked from
+	 * the Block's "more options" menu.
+	 *
+	 * @return {void}
+	 */
+	async function resizeContent() {
+		const block = select( blockEditorStore ).getSelectedBlock();
+		const blockContent = block.attributes.content ?? '';
+
+		setSelectedBlock( block );
+		setBlockContentAsPlainText( toPlainText( blockContent ) );
+		setIsResizing( true );
+	}
+
+	/**
+	 * The AJAX callback that returns with an array of suggestion.
+	 *
+	 * @return {Array} Array of suggestions.
+	 */
+	async function getResizedContent() {
+		let __textArray = [];
+		const apiUrl = `${ wpApiSettings.root }classifai/v1/openai/resize-content`;
+		const postId = select( editorStore ).getCurrentPostId();
+		const formData = new FormData();
+
+		formData.append( 'id', postId );
+		formData.append( 'content', blockContentAsPlainText );
+		formData.append( 'resize_type', resizingType );
+
+		dispatch( resizeContentStore ).setClientId( selectedBlock.clientId );
+
+		const response = await fetch( apiUrl, {
+			method: 'POST',
+			body: formData,
+			headers: new Headers( {
+				'X-WP-Nonce': wpApiSettings.nonce,
+			} ),
+		} );
+
+		if ( 200 === response.status ) {
+			__textArray = await response.json();
+		} else {
+			setIsResizing( false );
+			dispatch( resizeContentStore ).setClientId( '' );
+			resetStates();
+		}
+
+		setIsResizing( false );
+		dispatch( resizeContentStore ).setClientId( '' );
+
+		return __textArray;
+	}
+
+	/**
+	 * Updates the text selection.
+	 *
+	 * @param {string} updateWith The content that will be used to replace the selection.
+	 */
+	function updateContent( updateWith ) {
+		dispatch( blockEditorStore ).updateBlockAttributes(
+			selectedBlock.clientId,
+			{
+				content: updateWith,
+			}
+		);
+
+		dispatch( blockEditorStore ).selectionChange(
+			selectedBlock.clientId,
+			'content',
+			0,
+			updateWith.length
+		);
+		resetStates();
+	}
+
+	// We don't want to use the reszing feature when multiple blocks are selected.
+	// Nor do we want to use it when a block's content resizing is under process.
+	if ( isMultiBlocksSelected || isResizing ) {
+		return null;
+	}
+
+	// Result Modal JSX.
+	const suggestionModal = ! isResizing && textArray.length && isModalOpen && (
+		<Modal
+			title={ __( 'Suggestions', 'classifai' ) }
```

Comment:
Personal opinion but right now the title of the modal is `Suggestions` and then right below that we show `Suggestion` in the table header. Seems redundant to have these one after the other. Wondering if we should change this title to something else, like `Review Suggestions`, `Choose a Suggestion`, `Review Text`, `Select a Suggestion`